### PR TITLE
RELATED: CQ-582 search button in top menu

### PR DESCRIPTION
**Pull request by andriichumak**
- Refactored search drop-down according to the new design

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by andriichumak** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2692/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

File: libs/sdk-ui-kit/src/Header/HeaderSearch.tsx [hunk 1]
```diff
@@ -1,20 +1,67 @@
...
 import cx from "classnames";
-import { FormattedMessage } from "react-intl";
+import { FormattedMessage, injectIntl, WrappedComponentProps } from "react-intl";
 import { Button } from "../Button/index.js";
+import { Overlay } from "../Overlay/index.js";
 
-type HeaderSearchProps = {
-    onClick: () => void;
+type HeaderSearchCoreProps = React.PropsWithChildren<{
+    // onClick: () => void;
```
Comment (xMort) [inline libs/sdk-ui-kit/src/Header/HeaderSearch.tsx:9]: forgotten comment

File: libs/sdk-ui/src/base/localization/bundles/en-US.json [hunk 2]
```diff
@@ -3217,5 +3217,15 @@
...
         "limit": 0
+    },
+    "semantic-search.error": {
+        "value": "Search failed. Please try again later.",
+        "comment": "Error message shown when semantic search failed.",
+        "limit": 0
+    },
+    "semantic-search.no-results": {
+        "value": "No results found for “{query}”.",
+        "comment": "Message shown when semantic search did not return any results.",
```
Comment (xMort) [inline libs/sdk-ui/src/base/localization/bundles/en-US.json:3228]: We usually comment about placeholder that should not be translated.

**Review by ivanmjartan — APPROVED** [review]

**Review by ivanmjartan — APPROVED** [review]

**Pull request merged**